app/src/main/java/com/vestrel00/daggerbutterknifemvp/ui/common/BaseActivityModule.java

```diff
@@ -0,0 +1,34 @@
+package com.vestrel00.daggerbutterknifemvp.ui.common;
+
+import android.app.Activity;
+import android.app.FragmentManager;
+import android.content.Context;
+
+import com.vestrel00.daggerbutterknifemvp.inject.PerActivity;
+
+import javax.inject.Named;
+
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+
+/**
+ * Provides base activity dependencies. This must be included in all activity modules, which must
+ * provide a concrete implementation of {@link Activity}.
+ */
+@Module
+public abstract class BaseActivityModule {
+
+    static final String ACTIVITY_FRAGMENT_MANAGER = "BaseActivityModule.activityFragmentManager";
+
+    @Binds
+    @PerActivity
+    abstract Context activityContext(Activity activity);
```

Comment:
Scoping the `Context activityContext(Activity activity)` with `@PerActivity` is not necessary since the `Activity` instance will always be unique (new instances of it will not be created even without any scope). In general, providing `Application`, `Activity`, `Fragment`, `Service`, etc does not require scoped annotations since they are the components being injected and their instance is unique.

The same thing applies to `static FragmentManager activityFragmentManager(Activity activity)`. The `Activity` instance is unique so the `FragmentManager` it returns always come from the same Activity. Thus, the `@PerActivity` is not necessary here as the scope is implicitly per activity (literally).

However, using scope annotations in these cases makes the module easier to read. We wouldn’t have to look at what is being provided in order to understand its scope. I choose [readability here over (negligible) “performance/optimization”](https://github.com/google/dagger/issues/832#issuecomment-320510038).